freqtrade/freqai/data_drawer.py

```diff
@@ -423,10 +430,12 @@ def save_data(self, model: Any, coin: str, dk: FreqaiDataKitchen) -> None:
         save_path = Path(dk.data_path)
 
         # Save the trained model
-        if not dk.keras:
+        if self.model_type == 'joblib':
             dump(model, save_path / f"{dk.model_filename}_model.joblib")
-        else:
+        elif self.model_type == 'keras':
             model.save(save_path / f"{dk.model_filename}_model.h5")
+        elif 'stable_baselines' in self.model_type:
+            model.save(save_path / f"{dk.model_filename}_model.zip")
```

Comment:
This is somewhat broken logic (don't ask me why).

`freqtrade backtesting --config config_examples/config_freqai.example.json --strategy FreqaiExampleStrategy --freqaimodel LightGBMRegressor --strategy-path freqtrade/templates --timerange 20221001-20221005` fails if the configuration has an "rl_config" segment inside, so the config is somewhat leaking - making experimentation with different models (but a similar configuration) impossible.

Reply:
I fixed this - now the data drawer initializer will look for `Reinforcement` in the `freqaimodel` and if so, it will automatically switch the `model_type`. This is not a full-proof method - but it will help the users who use the FreqAI provided prediction models.